src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -2,80 +2,261 @@
-       // remove the following line when done
-       return null;
+       List<EnrollmentDTO> transcript = new ArrayList<EnrollmentDTO>();
+       List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsByStudentIdOrderByTermId(student.getId());
+
+       for (Enrollment e : enrollments) {
+           transcript.add(
+                new EnrollmentDTO(
+                    e.getEnrollmentId(),
+                    e.getGrade(),
+                    e.getStudent().getId(),
+                    e.getStudent().getName(),
+                    e.getStudent().getEmail(),
+                    e.getSection().getCourse().getCourseId(),
+                    e.getSection().getCourse().getTitle(),
+                    e.getSection().getSecId(),
+                    e.getSection().getSectionNo(),
+                    e.getSection().getBuilding(),
+                    e.getSection().getRoom(),
+                    e.getSection().getTimes(),
+                    e.getSection().getCourse().getCredits(),
+                    e.getSection().getTerm().getYear(),
+                    e.getSection().getTerm().getSemester()
+                )
+            );
+       }
+       return transcript;
    }
 
     // student gets a list of their enrollments for the given year, semester
     // user must be student
     // studentId will be temporary until Login security is implemented
    @GetMapping("/enrollments")
+   @PreAuthorize("hasAuthority('SCOPE_ROLE_STUDENT')")
    public List<EnrollmentDTO> getSchedule(
-           @RequestParam("year") int year,
-           @RequestParam("semester") String semester,
-           @RequestParam("studentId") int studentId) {
+       @RequestParam("year") int year,
+       @RequestParam("semester") String semester,
+       Principal principal) {
 
+       User student = userRepository.findByEmail(principal.getName());
 
-     // TODO
-	 //  hint: use enrollment repository method findByYearAndSemesterOrderByCourseId
-     //  remove the following line when done
-       return null;
-   }
+       if (student == null){
+           throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "user not found");
+       }
+       if (!student.getType().equals("STUDENT")){
+           throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "not authorized for this user");
+       }
 
+       List<EnrollmentDTO> schedule = new ArrayList<>();
+       List<Enrollment> enrollments = enrollmentRepository.findByYearAndSemesterOrderByCourseId(year, semester, student.getId());
+       //we are not checking for empty enrollments/ enrollments not found
+       for (Enrollment e : enrollments) {
+
+           if (!e.getStudent().getType().equals("STUDENT")) {
+               throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "user is not a student");
+           } else if (student.getId() != e.getStudent().getId()) {
+               throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "student id does not match");
+           }
```

Comment:
Probably doesn't hurt to leave this, but these should be already checked for since we uses preauthorize that the role is student, and we used the students id to retrieve the ernrollments.